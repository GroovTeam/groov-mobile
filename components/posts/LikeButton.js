import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import IconToggle from '../IconToggle';

const LikeStyles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  }
});

const LikeButton = ({ onLike, onUnlike, likeCount, alreadyLiked }) => {
  return (
    <View style={LikeStyles.container}>
      <IconToggle
        onActivate={onLike}
        onDeactivate={onUnlike}
        onIcon={'heart'}
        offIcon={'heart-outline'}
        color={'#dc143c'}
        startStatus={alreadyLiked}
        size={20}
      />
      <Text style={{fontWeight: '200', fontSize: 15, marginLeft: 1}}>
        {likeCount}
      </Text>
    </View>
  );
};

export default LikeButton;