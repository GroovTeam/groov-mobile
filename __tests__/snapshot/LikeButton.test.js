import React from 'react';
import { cleanup, render } from '@testing-library/react-native';

import LikeButton from '../../components/posts/LikeButton';

describe('<LikeButton />', () => {
  afterEach(cleanup);

  it('renders properly', () => {
    const tree = render(
      <LikeButton
        username={'test'}
        likes={['one', 'two']}
        likeCount={2}
        size={32}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});