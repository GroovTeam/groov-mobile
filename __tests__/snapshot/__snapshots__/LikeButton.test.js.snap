// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikeButton /> renders properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": 5,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text />
  </View>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "200",
        "marginLeft": 1,
      }
    }
  >
    2
  </Text>
</View>
`;
